lib/extract_field/FieldExtractorManager.js

```diff
@@ -0,0 +1,40 @@
+const FieldExtractor = require("perimeterx-node-core/lib/extract_field/FieldExtractor");
+const ExtractionField = require("perimeterx-node-core/lib/models/ExtractionField");
+
+const USERNAME_FIELD = "user";
+const PASSWORD_FIELD = "pass";
+
+class FieldExtractorManager { 
+    constructor(extractObjects) {
+        this.extractorMap = this._createExtractorsMap(extractObjects);
+    }
+
+    ExtractFields(request) {
+        const key = this._getMapKey(request.path, request.method);
```

Comment:
don't you need request.method.toUpperCase() here as well?

Reply:
HTTP method comes in as uppercase always